source-braintree-native/source_braintree_native/api.py

```diff
@@ -0,0 +1,261 @@
+                    elif isinstance(item, datetime):
+                        new_value.append(item.replace(tzinfo=UTC))
+                    else:
+                        new_value.append(item)
+
+                data[key] = new_value
+
+        # Remove private attributes.
+        data.pop('_setattrs', None)
+        return data
+
+
+async def snapshot_resources(
+        braintree_gateway: BraintreeGateway,
+        gateway_property: str,
+        gateway_response_field: str | None,
+        log: Logger,
+) -> AsyncGenerator[FullRefreshResource, None]:
+    resources = getattr(braintree_gateway, gateway_property).all()
+
+    iterator = getattr(resources, gateway_response_field) if gateway_response_field else resources
+    for object in iterator:
+        yield FullRefreshResource.model_validate(_braintree_object_to_dict(object))
+
+
+async def fetch_transactions(
+        braintree_gateway: BraintreeGateway,
+        window_size: int,
+        log: Logger,
+        log_cursor: LogCursor,
+) -> AsyncGenerator[IncrementalResource | LogCursor, None]:
+    assert isinstance(log_cursor, datetime)
+    most_recent_created_at = log_cursor
+    window_end = log_cursor + timedelta(days=window_size)
+    end = min(window_end, datetime.now(tz=UTC))
+
+    collection = braintree_gateway.transaction.search(
+        TransactionSearch.created_at.between(log_cursor, end)
+    )
+
+    count = 0
+
+    for object in collection.items:
+        count += 1
+        doc = IncrementalResource.model_validate(_braintree_object_to_dict(object))
+
+        if doc.created_at > log_cursor:
+            yield doc
+            most_recent_created_at = doc.created_at
+
+    if count >= TRANSACTION_SEARCH_LIMIT:
+        raise RuntimeError(_search_limit_error_message(count, "transactions"))
+
+    if end == window_end:
+        yield window_end
+    elif most_recent_created_at > log_cursor:
+        yield most_recent_created_at
+
+
+async def fetch_customers(
+        braintree_gateway: BraintreeGateway,
+        window_size: int,
+        log: Logger,
+        log_cursor: LogCursor,
+) -> AsyncGenerator[IncrementalResource | LogCursor, None]:
+    assert isinstance(log_cursor, datetime)
+    most_recent_created_at = log_cursor
+    window_end = log_cursor + timedelta(days=window_size)
+    end = min(window_end, datetime.now(tz=UTC))
+
+    collection = braintree_gateway.customer.search(
+        CustomerSearch.created_at.between(log_cursor, end)
+    )
+
+    count = 0
+
+    for object in collection.items:
+        count += 1
+        doc = IncrementalResource.model_validate(_braintree_object_to_dict(object))
+
+        if doc.created_at > log_cursor:
+            yield doc
+            most_recent_created_at = doc.created_at
+
+    if count >= SEARCH_LIMIT:
+        msg = f"{count} customers returned in a single search which is greater than or equal to Braintree's documented maximum for a single customer search. Reduce the window size and backfill this stream."
```

Comment:
Is there a reason this error message is still here instead of using `_search_limit_error_message` like the other generator functions?

Reply:
Nope, I overlooked it when I was refactoring. Thanks for catching that! I'll update it before merging.